# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeChild.cpp
# Commit: 71b6060b6015
# Full Hash: 71b6060b60151e157e94110d28fb51d76bb6415a
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-12-08 04:18:28
# Regressor Bug: 1863914
# File Overlap Count: 2
# Description:
#   Bug 1863914: Use multiple shmem buffers for remote canvas recording. r=aosmond
#   
#   This replaces the use of a single large ring buffer.
#   The buffers are still processed in parallel and are recycled to reduce
#   allocation. Events that do not fit in the default sized buffer have a separate
# ==============================================================================

diff -r bf28b3c066e6 -r 71b6060b6015 gfx/layers/ipc/CompositorBridgeChild.cpp
--- a/gfx/layers/ipc/CompositorBridgeChild.cpp	Thu Dec 07 17:10:06 2023 +0000
+++ b/gfx/layers/ipc/CompositorBridgeChild.cpp	Thu Dec 07 17:27:24 2023 +0000
@@ -531,7 +531,9 @@
 }
 
 void CompositorBridgeChild::ClearCachedResources() {
-  CanvasChild::ClearCachedResources();
+  if (auto* cm = gfx::CanvasManagerChild::Get()) {
+    cm->ClearCachedResources();
+  }
 }
 
 bool CompositorBridgeChild::AllocUnsafeShmem(size_t aSize, ipc::Shmem* aShmem) {